File: DSA/Sarthak_Jain_590014936/DAY_1/Day_1_Question_1.py
#Sarthak Jain 590014936
#Day 1 Question 1
#Replace All Negative Numbers with Zero

lis=[]

a=int(input("Enter number of elements:"))
for i in range(a):
    b=int(input("Enter element:"))
    lis.append(b)

print("Given List:")
print(lis)

for x in range(a):
    if lis[x]<0:
        lis[x]=0

print("Updated list with Negative numbers replaced by 0:")
print(lis)

#Basic array manipulation by checking every element at indexes of the list and changing if <0

File: DSA/vrishali_sahay_590011701/Day1_Question1.c
#include <stdio.h>

// Replace all negative numbers in the array with zero
void replaceNegativesWithZero(int nums[], int n) {
    for (int i = 0; i < n; i++) {
        if (nums[i] < 0) {
            nums[i] = 0;
        }
    }
}

int main() {
    int n;
    printf("Enter the size of the array: ");
    scanf("%d", &n);

    int nums[n];
    printf("Enter %d elements:\n", n);
    for (int i = 0; i < n; i++) {
        scanf("%d", &nums[i]);
    }

    replaceNegativesWithZero(nums, n);

    printf("Modified array:\n");
    for (int i = 0; i < n; i++) {
        printf("%d ", nums[i]);
    }
    printf("\n");

    return 0;
}
